### PR TITLE
Add style check and also fix style violations

### PR DESCRIPTION
**Pull request by kelloggm**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -36,3 +39,12 @@ publishing {
         }
     }
 }
+
+// run google java format to enforce style
+spotless {
+    // uncomment this line to temporarily disable spotless (i.e. when debugging)
+    // enforceCheck = false
+    java {
+        googleJavaFormat()
```
Comment (mernst) [inline build.gradle]: google-java-format produces [bad formatting](https://github.com/google/google-java-format/issues/5) for type annotations.
Therefore, I don't think its use should be encouraged, in projects related to type annotations, until https://github.com/google/google-java-format/issues/5 is fixed.

File: src/test/java/tests/TemplateforaTest.java [hunk 2]
```diff
@@ -13,18 +13,18 @@
...
-    public TemplateforaTest(List<File> testFiles) {
-        super(
-                testFiles,
-                TemplateforaChecker.class,
-                "templatefora",
-                "-Anomsgtext",
-                "-Astubs=stubs/",
-                "-nowarn");
-    }
+  public TemplateforaTest(List<File> testFiles) {
```
Comment (mernst) [inline src/test/java/tests/TemplateforaTest.java]: The Checker Framework uses the `-a` command-line argument to google-java-format, for 4-space indentation.
I dislike that, but we might want to keep this project consistent with the other Checker Framework repositories.

File: build.gradle (hunk 1 above)
Comment (kelloggm) [reply build.gradle]: Do you have a suggestion for a style checker that does not have bad formatting for type annotations? I'm not aware of one.

Comment (mernst) [reply build.gradle]: https://github.com/plume-lib/run-google-java-format
It's not as convenient to use (ugly Gradle integration), but it formats correctly.

**Pull request closed without merging**